src/app/shared-services/inferences/inference.service.ts

```diff
@@ -29,7 +31,8 @@ export class InferenceFilterConfiguration {
 })
 export class InferenceService implements OnDestroy {
   private static inferenceIntervalMinutes = 240 // 4 hours
-  private inferenceEngine = new SimpleEngine()
+  // private inferenceEngine = new SimpleEngine()
```

Comment:
Maybe you could add a flag that determines the used engine? (Ideally we even would some sort of `InferenceEngineFactory`, that simply returns an `InferenceEngine` to this service – `InferenceService` doesn't need to know, what kind of engine in particular it gets.)

Reply:
Agreed, I went with the flag option for now to not introduce too much additional complexity into this already quite elaborate inference engine architecture.